boards/riscv32/rv32m1_vega/rv32m1_vega_ri5cy.dts

```diff
@@ -16,6 +16,7 @@
 		zephyr,sram = &m4_dtcm;
 		zephyr,flash = &ri5cy_code_partition;
 		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
 		zephyr,uart-pipe = &uart0;
```

Comment:
Is it really OK to have this share the same peripheral with zephyr,uart-pipe? The `CONFIG_UART_PIPE` help makes it seem like this should be a different peripheral than the console/shell.

Reply:
Valid question.

However, it may be orthogonal to this change.  Without which, CONFIG_UART_SHELL_ON_DEV_NAME is empty for any sample that uses CONFIG_SHELL=y.

Btw, if you're right and uart-pipe needs to be different, then almost every board in Zephyr is configured wrong.  Example K64F: https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/frdm_k64f/frdm_k64f.dts#L45

Reply:
As a follow up, here is the usage of each of these items:
```
zephyr/drivers/console/uart_console.c:616:	uart_console_dev = device_get_binding(CONFIG_UART_CONSOLE_ON_DEV_NAME);
zephyr/subsys/shell/shell_uart.c:270:			device_get_binding(CONFIG_UART_SHELL_ON_DEV_NAME);
zephyr/drivers/console/uart_pipe.c:94:	uart_pipe_dev = device_get_binding(CONFIG_UART_PIPE_ON_DEV_NAME);
```

Here's the usage matrix:
```
UART_CONSOLE_ON_DEV_NAME depends on UART_CONSOLE || CONSOLE_SUBSYS
UART_PIPE_ON_DEV_NAME depends on UART_PIPE which depends on UART_INTERRUPT_DRIVEN
UART_SHELL_ON_DEV_NAME requires SHELL_BACKEND_SERIAL
```

From the looks of it, you can enable all 3 at the same time and they will all attempt to use the same device.   Or maybe the intended use is to let the sample enable which ever is needed and they all happen to point to the same device, but really aren't intended to be used at once.

I seem to remember getting double output at one point and having it explained that I had CONSOLE and SHELL both using the same UART.

Reply:
Give that there are multiple examples where the same name is used it seems to me that there is an expectation of mutual exclusion between the devices being enabled at the same time. Maybe that is what is needed?

Reply:
> Give that there are multiple examples where the same name is used it seems to me that there is an expectation of mutual exclusion between the devices being enabled at the same time. Maybe that is what is needed?

Maybe? Why is the pipe Kconfig in drivers/console/Kconfig though, then? I feel like I don't really understand what it's for now.